core/src/test/java/io/questdb/log/LogFactoryTest.java

```diff
@@ -662,6 +664,61 @@ public void testWhenCustomLogLocationIsSpecifiedThenDefaultLogFileIsNotUsed() th
         testCustomLogIsCreated(false);
     }
 
+    @Test
+    public void testEnvVarPriorityOverConfigFileSettings() throws IOException {
+        File conf = temp.newFile();
+        File out = new File(temp.newFolder(), "testSetProperties.log");
+
+        TestUtils.writeStringToFile(conf, "writers=file\n" +
+                "recordLength=4096\n" +
+                "queueDepth=1024\n" +
+                "w.file.class=io.questdb.log.LogFileWriter\n" +
+                "w.file.location=" + out.getAbsolutePath().replaceAll("\\\\", "/") + "\n" +
+                "w.file.level=INFO,ERROR\n" +
+                "w.file.bufferSize=4M");
+
+        Map<String, String> env = new HashMap<>();
+        // First configure a LogFactory using the above config file and an empty env
+        System.setProperty(LogFactory.CONFIG_SYSTEM_PROPERTY, conf.getAbsolutePath());
+        try (LogFactory factory = new LogFactory()) {
+            LogFactory.configureFromSystemProperties(factory, null, env, false);
+
+            Log logger = factory.create("xyz");
+            assertDisabled(logger.debug());
+            assertEnabled(logger.info());
+            assertEnabled(logger.error());
+            assertEnabled(logger.advisory());
+        }
+
+        // Now override a setting in the environment and retest
+        // todo: Figure out why setting to DEBUG only enables DEBUG and disables all
+        // other levels
+        env.put("QDB_LOG_W_FILE_LEVEL", "DEBUG");
```

Comment:
@bziobrowski this may be a bug? I can't figure it out from the code, which is here:
```
        // enable all LOG levels above the minimum set one
        // ((-1 >>> (msb-1)) << msb) | level
        final int msb = Numbers.msb(level);
        level = (((-1 >>> (msb - 1)) << msb) | level) & LogLevel.MASK;
```